plugins/redis/redisplugin.hpp

```diff
@@ -0,0 +1,134 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef ECL_REDIS_INCL
+#define ECL_REDIS_INCL
+
+#ifdef _WIN32
+#define ECL_REDIS_CALL _cdecl
+#ifdef ECL_REDIS_EXPORTS
+#define ECL_REDIS_API __declspec(dllexport)
+#else
+#define ECL_REDIS_API __declspec(dllimport)
+#endif
+#else
+#define ECL_REDIS_CALL
+#define ECL_REDIS_API
+#endif
+
+#include "jhash.hpp"
+#include "hqlplugins.hpp"
+#include "eclhelper.hpp"
+#include "jexcept.hpp"
+#include "hiredis/hiredis.h"
+
+extern "C"
+{
+    ECL_REDIS_API bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb);
+    ECL_REDIS_API void setPluginContext(IPluginContext * _ctx);
+}
+
+class StringBuffer;
+
+namespace RedisPlugin {
+static const unsigned unitExpire = 86400;//1 day (secs)
+
+#define setFailMsg "'Set' request failed - "
+#define getFailMsg "'Get<type>' request failed - "
+
+StringBuffer & appendExpire(StringBuffer & buffer, unsigned expire);
+
+class RedisServer : public CInterface
+{
+public :
+    RedisServer() : port(0) { }
+    RedisServer(ICodeContext * ctx, const char * _options)
+    {
+        serverPswdHash = hashc((const unsigned char*)_options, strlen(_options), 0);
+        options.set(_options);
+        parseOptions(ctx, _options);
+    }
+    RedisServer(ICodeContext * ctx, const char * _options, const char * pswd)
+    {
+        serverPswdHash = hashc((const unsigned char*)_options, strlen(_options), serverPswdHash);
```

Comment:
I prefer 

variable = f(a, 0);
variable = f(b, variable);

versus nesting - easier to read.